youngjoo00/week_01/BOJ_1026_보물.swift

```diff
@@ -0,0 +1,24 @@
+// BOJ_1026_보물.swift
+//첫째 줄에 N이 주어진다. 둘째 줄에는 A에 있는 N개의 수가 순서대로 주어지고, 셋째 줄에는 B에 있는 수가 순서대로 주어진다.
+//N은 50보다 작거나 같은 자연수이고, A와 B의 각 원소는 100보다 작거나 같은 음이 아닌 정수이다.
+
+let N = Int(readLine()!)!
+var A : [Int] = readLine()!.split(separator: " ").map { Int($0)! }.sorted()
+var B : [Int] = readLine()!.split(separator: " ").map { Int($0)! }
+
+var result = 0
+
+// 오름차순으로 정렬된 A 배열의 i * B의 max값 을 result에 더하면 끝
+// B의 max 값은 반복횟수마다 remove 로 배열에서 값을 날려버림
+for i in A {
+    result += i * B.max()!
+    B.remove(at: B.firstIndex(of: B.max()!)!)
+}
```

Comment:
swift의 max와 remove는 모두 O(n)이라는 시간복잡도를 가지니까
for문에서 1번 반복할 때마다 O(2n)의 시간복잡도에요
```suggestion
let N = Int(readLine()!)!
var A : [Int] = readLine()!.split(separator: " ").map { Int($0)! }.sorted()
var B : [Int] = readLine()!.split(separator: " ").map { Int($0)! }.sorted(by : >) #B를 내림차순으로 정렬

var result = 0

// 오름차순으로 정렬된 A 배열의 i * B의 max값 을 result에 더하면 끝
// B의 max 값은 반복횟수마다 remove 로 배열에서 값을 날려버림
for i in 0..<A.count {
    result +=A[i]*B[i]!
}
```
한 번의 정렬을 추가하는건 nlogn이니까 위의 코드처럼 써보는건 어떨까요?

Reply:
아주 기가맥히는 발상이었읍니다